packages/blade/src/components/Button/Button/__tests__/Button.web.test.tsx

```diff
@@ -0,0 +1,114 @@
+    const { container } = renderWithTheme(<Button size="xsmall">{buttonText}</Button>);
+    expect(container).toMatchSnapshot();
+  });
+
+  it('should render small size button', () => {
+    const buttonText = 'Pay Now';
+    const { container } = renderWithTheme(<Button size="small">{buttonText}</Button>);
+    expect(container).toMatchSnapshot();
+  });
+
+  it('should render medium size button', () => {
+    const buttonText = 'Pay Now';
+    const { container } = renderWithTheme(<Button size="medium">{buttonText}</Button>);
+    expect(container).toMatchSnapshot();
+  });
+
+  it('should render large size button', () => {
+    const buttonText = 'Pay Now';
+    const { container } = renderWithTheme(<Button size="large">{buttonText}</Button>);
+    expect(container).toMatchSnapshot();
+  });
+
+  it('should render button with icon with default iconPosition', () => {
+    const buttonText = 'Pay Now';
+    const { container } = renderWithTheme(<Button icon={CreditCardIcon}>{buttonText}</Button>);
+    expect(container).toMatchSnapshot();
+  });
+
+  it('should render button with icon with right iconPosition', () => {
+    const buttonText = 'Pay Now';
+    const { container } = renderWithTheme(
+      <Button icon={CreditCardIcon} iconPosition="right">
+        {buttonText}
+      </Button>,
+    );
+    expect(container).toMatchSnapshot();
+  });
+
+  it('should render button with full width', () => {
+    const buttonText = 'Pay Now';
+    const { container } = renderWithTheme(<Button isFullWidth={true}>{buttonText}</Button>);
+    expect(container).toMatchSnapshot();
+  });
+
+  it('should render disabled button', () => {
+    const buttonText = 'Pay Now';
+    const { container, getByRole } = renderWithTheme(
+      <Button isDisabled={true}>{buttonText}</Button>,
+    );
+    expect(container).toMatchSnapshot();
+    expect(getByRole('button')).toBeDisabled();
+  });
+
+  it('should render secondary variant button', () => {
+    const buttonText = 'Pay Now';
+    const { container } = renderWithTheme(<Button variant="secondary">{buttonText}</Button>);
+    expect(container).toMatchSnapshot();
+  });
+
+  it('should render disabled secondary variant button', () => {
+    const buttonText = 'Pay Now';
+    const { container } = renderWithTheme(
+      <Button variant="secondary" isDisabled={true}>
+        {buttonText}
+      </Button>,
+    );
+    expect(container).toMatchSnapshot();
+  });
+
+  it('should render tertiary variant button', () => {
+    const buttonText = 'Pay Now';
+    const { container } = renderWithTheme(<Button variant="tertiary">{buttonText}</Button>);
+    expect(container).toMatchSnapshot();
+  });
+
+  it('should render disabled tertiary variant button', () => {
+    const buttonText = 'Pay Now';
+    const { container } = renderWithTheme(
+      <Button variant="tertiary" isDisabled={true}>
+        {buttonText}
+      </Button>,
+    );
+    expect(container).toMatchSnapshot();
```

Comment:
for web can we add assertion to check that the button is not clickable in disabled state?
```
const button = getByRole('button');
fireEvent.click(button);
expect(onClick).toHaveBeenCalledTimes(0);
```